app/src/main/java/campus/tech/kakao/map/view/SearchPlaceActivity.kt

```diff
@@ -44,22 +43,27 @@ class SearchPlaceActivity : AppCompatActivity() {
 
         viewModel.updateSavedSearch()
 
+
+
+
         //-----viewModel observe-----------------------------------------
         val activity = this
         with(viewModel) {
 
             //PlaceAdapter
             itemClick.observe(activity, Observer { place ->  //Place 클릭 했을 때
-//                setSharedPreferences(place)
+                val resultIntent = Intent()
+                resultIntent.putExtra("name", place.name)
+                resultIntent.putExtra("address", place.address)
+                resultIntent.putExtra("latitude", place.latitude)
+                resultIntent.putExtra("longitude", place.longitude)
+                setResult(Activity.RESULT_OK, resultIntent)
```

Comment:
👍 👍 👍 Intent를 이용해 이전화면으로 정보를 돌려줄 수 있도록 잘 작성해주셨네요!

조금 아쉬운 점은 "name", "address"와 같은 키값을 하드코딩하고 있다는 점이예요. MainActivity가 다시 되돌려받을 때(getExtra)도 이 키값을 써야하는데, 하드코딩하게되면 오타가 생길 수 있으니까요. Activity에 const 키워드로 상수 선언해주시는 것이 좋습니다.

그리고 조금 더 깔끔하게 다듬어보자면 정의하셨던 Location 클래스를 이용해 Intent에 담아주셨으면 합니다. `Intent`와 `Parcelable`을 키워드로 한번 학습해보시면 좋겠어요.

```suggestion
                val resultIntent = Intent()
                resultIntent.putExtra("location", place)
                setResult(Activity.RESULT_OK, resultIntent)
```